src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/ImplementGenericMathInterfacesCorrectly.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.NetCore.Analyzers.Usage
+{
+    using static MicrosoftNetCoreAnalyzersResources;
+
+    public abstract class ImplementGenericMathInterfacesCorrectly : DiagnosticAnalyzer
+    {
+        private const string RuleId = "CA2260";
+
+        internal static readonly DiagnosticDescriptor GMInterfacesRule = DiagnosticDescriptorHelper.Create(
+            RuleId,
+            CreateLocalizableResourceString(nameof(ImplementGenericMathInterfacesCorrectlyTitle)),
+            CreateLocalizableResourceString(nameof(ImplementGenericMathInterfacesCorrectlyMessage)),
+            DiagnosticCategory.Usage,
+            RuleLevel.BuildWarning,
+            description: CreateLocalizableResourceString(nameof(ImplementGenericMathInterfacesCorrectlyDesciption)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        private static readonly ImmutableArray<string> s_knownInterfaces = ImmutableArray.Create("IParsable`1", "ISpanParsable`1", "IAdditionOperators`3", "IAdditiveIdentity`2",
+            "IBinaryFloatingPointIeee754`1", "IBinaryInteger`1", "IBinaryNumber`1", "IBitwiseOperators`3", "IComparisonOperators`2", "IDecrementOperators`1", "IDivisionOperators`3",
+            "IEqualityOperators`2", "IExponentialFunctions`1", "IFloatingPointIeee754`1", "IFloatingPoint`1", "IHyperbolicFunctions`1", "IIncrementOperators`1", "ILogarithmicFunctions`1",
+            "IMinMaxValue`1", "IModulusOperators`3", "IMultiplicativeIdentity`2", "IMultiplyOperators`3", "INumberBase`1", "INumber`1", "IPowerFunctions`1", "IRootFunctions`1", "IShiftOperators`2",
+            "ISignedNumber`1", "ISubtractionOperators`3", "ITrigonometricFunctions`1", "IUnaryNegationOperators`2", "IUnaryPlusOperators`2", "IUnsignedNumber`1", "IFloatingPointConstants`1");
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(GMInterfacesRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemIParsable1, out var iParsableInterface) ||
+                    !context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemNumericsIDecrementOperators1, out var iBinaryInteger1))
```

Comment:
why `WellKnownTypeNames.SystemNumericsIDecrementOperators1` but `out var iBinaryInteger1`?

If the type isn't used, could we just `out _` instead?

Maybe `INumberBase` is a better one to check for?

Reply:
Also why only these two? Do we not need to register for all of the interfaces?

Reply:
I did not wanted to load all interfaces, instead keep all interface names, all interfaces in the list are in System and System.Numerics namespace. So this 2 used for checking:
1. If the interfaces available in the compilation
2. if an interface name found in the list we check if it is from System or System.Numerics namespace

Reply:
> why WellKnownTypeNames.SystemNumericsIDecrementOperators1 but out var iBinaryInteger1?

I was using iBinaryInteger1 at first 😆, that time I was using mock interfaces as the real interfaces was not available in the repo, and IBinaryInteger was derived from other interfaces, IDecrementOperator was simpler to mock

> Maybe INumberBase is a better one to check for?

Now as the repo SDK is using preview 7 we can use any, probably better to be the one that added with 1st release

Reply:
`INumber` was one of the first added and its really the "core" interface we expect most users will end up targeting.